File: 프로그래머스/unrated/161989. 덧칠하기/덧칠하기.java
class Solution {
    public int solution(int n, int m, int[] section) {
        int answer = 1;
        int roller = section[0];
        
        for (int wall : section) {
            if (roller + m <= wall) {
                roller = wall;
                answer++;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.js
function solution(n, m, section) {
    var current = section[0];
    var count = 1;
    for(var sec of section) {
        if (sec - current + 1 > m) {
            count += 1;
            current = sec;
        }
    }
    return count;
}


